credstash.py

```diff
@@ -259,19 +259,28 @@ def putSecret(name, secret, version="", kms_key="alias/credstash",
 
 
 def getAllSecrets(version="", region=None, table="credential-store",
-                  context=None, **kwargs):
+                  context=None, session=None, **kwargs):
     '''
     fetch and decrypt all secrets
     '''
     output = {}
     secrets = listSecrets(region, table, **kwargs)
+
+    # We only need these things once
+    if session is None:
+        session = get_session(**kwargs)
```

Comment:
`get_session` has caching mechanism
https://github.com/l2ol33rt/credstash/blob/master/credstash.py#L366-L374

No need to repeat it.